bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -527,42 +547,51 @@ private ByteBuf doGetEntry(long ledgerId, long entryId) throws IOException, Book
 
         dbLedgerStorageStats.getWriteCacheMissCounter().inc();
 
-        // Try reading from read-ahead cache
-        entry = readCache.get(ledgerId, entryId);
-        if (entry != null) {
-            dbLedgerStorageStats.getReadCacheHitCounter().inc();
-            return entry;
-        }
+        // Get entry from storage and trigger read-ahead
+        long readAheadTotalStartNano = MathUtils.nowInNano();
+        if (enableReadAheadAsync) {
+            // Async mode
+            entry = readAheadManager.readEntryOrWait(ledgerId, entryId);
+        } else {
+            // Sync mode
```

Comment:
could you encapsulate the original sync mode implementation into one meaningful method?